xAPI-Communication.md

```diff
@@ -1667,246 +1667,18 @@ described in [Appendix A: Converting Statements to 1.0.0](#Appendix3A).
-	<tr>
-		<td>state</td>
-		<td>read/write state data, limited to Activities and Actors 
-			associated with the current token to the extent it is 
-			possible to determine this relationship.
-		</td>
-	</tr>
-	<tr id="4.2.s2.table1.row4">
-		<td>define</td>
-		<td>(re)Define Activities and Actors. If storing a Statement 
-			when this is not granted, ids will be saved and the LRS 
-			MAY save the original Statement for audit purposes, but 
-			SHOULD NOT update its canonical representation of any 
-			Actors or Activities.
-		</td>
-	</tr>
-	<tr id="4.2.s2.table1.row5">
-		<td>profile</td>
-		<td>read/write Profile document data, limited to Activities and Actors 
-			associated with the current token to the extent it is 
-			possible to determine this relationship.
-		</td>
-	</tr>
-	<tr id="4.2.s2.table1.row6"><td>all/read</td><td>unrestricted read access</td></tr>
-	<tr id="4.2.s2.table1.row7"><td>all</td><td>unrestricted access</td></tr>
-</table>
-
-###### <a name="4.2.s3"></a>OAuth Resources
-<table>
-	<tr>
-		<th>Name</th>
-		<th>Endpoint</th>
-		<th>Example</th>
-	</tr>
-	<tr id="4.2.s3.table1.row1">
-		<td>Temporary Credential Request</td>
-		<td>OAuth/initiate</td>
-		<td>http://example.com/xAPI/OAuth/initiate</td>
-	</tr>
-	<tr id="4.2.s3.table1.row2">
-		<td>Resource Owner Authorization</td>
-		<td>OAuth/authorize</td>
-		<td>http://example.com/xAPI/OAuth/authorize</td>
-	</tr>
-	<tr id="4.2.s3.table1.row3">
-		<td>Token Request</td>
-		<td>OAuth/token</td>
-		<td>http://example.com/xAPI/OAuth/token </td>
-	</tr>
-</table>
-
-##### <a name="4.2.s4"></a>Example
-The list of scopes determines the set of permissions that is being requested. For example, an instructor might grant 
-"statements/read" to an application (Client), but the LRS would still limit that tool to Statements that the instructor could 
-read if querying the LRS with their credentials directly (such as Statements relating to their students).
-
-##### <a name="4.2.s5"></a>Requirements
-
-* <a name="4.2.s5.b1"></a>The LRS MUST accept a scope parameter as defined in [OAuth 2.0](http://tools.ietf.org/html/rfc6749#section-3.3).
-* <a name="4.2.s5.b2"></a>The LRS MUST assume a requested scope of "statements/write" and "statements/read/mine" if no 
-scope is specified.
-* <a name="4.2.s5.b3"></a>The LRS MUST support the scope of "all" as a minimum.
-* <a name="4.2.s5.b4"></a>The LRS MAY support other scopes.
-* <a name="4.2.s5.b5"></a>The Client SHOULD request only the minimal needed scopes, to increase the chances that the 
-request will be granted.
-* <a name="4.2.s5.b6"></a>The parameters "consumer_name" and "scope" are not part of OAuth 1.0, and therefore if used MUST be passed 
-as query string or form parameters, not in the OAuth header.  
+* <a name="4.0.s2.b1"></a>The LRS MUST handle making, or delegating, decisions on the validity of Statements, and 
+determining what operations might be performed based on the credentials used (or lack therof).
```

Comment:
This requirement seems hard to test. Do we need it as a requirement? Could it be descriptive text outlining that this is part of the LRS's role?

Reply:
perhaps this should be a `MAY` ? I think this is what gives the LRS permission to reject operations based on credentials used, but it does seem odd it's a `MUST`

Reply:
@FlorianTolk does "we need some measurable requirement" here mean you're opposed to descriptive text? "forces an LRS to protect the LRS from unauthorized sources trying to access it" seems a little odd because the LRS decides what is authorized and what is not, so is it really wrong for the LRS to decide that everyone is authorized? Making this descriptive of the LRS role (perhaps with a MAY statement as I suggested) seems in line with leaving security requirements in a follow on document.

It seems like it might be true that an LRS that considers all requests authorized might be unfit for many purposes, but does having a requirement here (one that won't be very testable) vs just noting the role of the LRS really help with that situation?

Reply:
Noted on today's call, this requirement is not new, it's already here but has moved from elsewhere. https://github.com/adlnet/xAPI-Spec/pull/1079/files#diff-01eaf9fd40bc44cd79621a7c76ebe3a5L1695